learning/scripting/c_sharp/c_sharp_features.rst

```diff
@@ -0,0 +1,72 @@
+.. _doc_c_sharp_features:
+
+Features
+============
+
+This page provied an overview over the commonly used features of both C# and Godot
+and how they are used together.
+
+Type Conversion and Casting
+---------------------------
+
+C# is a statically typed language. Therefore you can't do the following:
+
+.. code-block:: csharp
+
+    var my_sprite = get_node("MySprite")
+    my_sprite.set_frame(0)
+
+The method ``get_node()`` returns a ``Node`` instance.
+You must explicitly convert it to the desired derived type, ``Sprite`` in this case:
+
+.. code-block:: csharp
+
+    Sprite my_sprite = get_node("MySprite") as Sprite;
+    my_sprite.set_frame(0);
```

Comment:
Here I would explain it this way:

- Normal cast: `(Sprite)GetNode("MySprite")`. Throws `InvalidCastException` if the returned node cannot be casted to Sprite. You would use it instead of the `as` operator if you are pretty sure it won't fail.
- `as` operator: `GetNode("MySprite") as Sprite`. Returns null if the node cannot be casted to Sprite, and for this reason it cannot be used with value types.
- `is` operator: `GetNode("MySprite") is Sprite`. Returns false if the node cannot be casted to Sprite, otherwise true.

Finally, I would mention pattern matching and link to https://docs.microsoft.com/en-us/dotnet/csharp/pattern-matching for more details.